osf/migrations/0045_backfill_date_retracted.py

```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.9 on 2017-03-14 14:41
+
+"""
+Backfill Retraction.date_retracted with `RETRACTION_APPROVED` log date.
+"""
+
+from __future__ import unicode_literals
+
+from django.db import migrations
+
+from osf.models import Registration, Retraction, Sanction
+
+
+def forward(*args):
+    retractions = Retraction.objects.filter(state=Sanction.APPROVED, date_retracted=None)
+    total = retractions.count()
+
+    print('Migrating {} retractions.'.format(total))
+
+    for retraction in retractions:
+        parent_registration = Registration.objects.get(retraction=retraction)
+
+        retraction_logs = [
+            log for log in parent_registration.registered_from.logs.all()
```

Comment:
All of this filtering could be done in sql. Something like: `parent_registration.registered_from.logs.filter(action='retraction_approved', params__retraction_id=retraction._id)`